tests/test_mem_eff_attention.py

```diff
@@ -721,3 +724,85 @@ def test_memory_efficient_attention_full_block_masked(
     assert_allclose(grad_q, query.grad, "grad_q", atol=atol)
     assert_allclose(grad_k, key.grad, "grad_k", atol=atol)
     assert_allclose(grad_v, value.grad, "grad_v", atol=atol)
+
+
+@pytest.mark.parametrize("contiguous", [True, False])
+@pytest.mark.parametrize("dim", [0, 1, 2, 3, 4])
+def test_unbind(dim: int, contiguous: bool):
+    x = contiguous_x = torch.ones([10, 20, 4, 10, 3], requires_grad=True)
+    x2 = contiguous_x2 = torch.ones([10, 20, 4, 10, 3], requires_grad=True)
+
+    if not contiguous:
```

Comment:
nit: If you create a tensor that requires grad and then apply operations to it (overriding the original tensor), the tensor you'll store will be a derived tensor, and not the leaf tensor. That might be the reason why you had to store `contiguous_x` as well here.

What I would generally recommend is instead to create the tensor without `requires_grad=True`, perform the permutations / etc, and then make it require grad.
Something like
```python
x = torch.rand(...)
if not contiguous:
    x = x.permute(...)
x.requires_grad_(True)
```